packages/m2m-gateway/src/services/eserviceService.ts

```diff
@@ -171,6 +173,67 @@ export function eserviceServiceBuilder(
     };
   }
 
+  // eslint-disable-next-line max-params
+  async function deleteAttributeFromEServiceDescriptorGroup(
+    eserviceId: EServiceId,
+    descriptorId: DescriptorId,
+    groupIndex: number,
+    attributeId: AttributeId,
+    attributeKind: keyof catalogApi.Attributes,
+    { headers }: WithLogger<M2MGatewayAppContext>
+  ): Promise<void> {
+    const eservice = await retrieveEServiceById(headers, eserviceId);
+    const descriptor = retrieveEServiceDescriptorById(eservice, descriptorId);
+
+    const descriptorAttributes = descriptor.attributes[attributeKind];
+
+    const attributeGroup = descriptorAttributes[groupIndex];
```

Comment:
I think in this way it explodes runtime, while using `.at(` you get undefined in case the group does not exist
```suggestion
    const attributeGroup = descriptorAttributes.at(groupIndex);
```